qiskit/aqua/circuits/phase_estimation_circuit.py

```diff
@@ -141,11 +148,21 @@ def construct_circuit(
             if auxiliary_register is None:
                 num_aux_qubits, aux = 0, None
                 if self._state_in_circuit_factory is not None:
-                    num_aux_qubits = self._state_in_circuit_factory.required_ancillas()
+                    if isinstance(self._state_in_circuit_factory, CircuitFactory):
+                        num_aux_qubits = self._state_in_circuit_factory.required_ancillas()
+                    elif hasattr(self._state_in_circuit_factory, 'num_ancilla_qubits'):
+                        num_aux_qubits = self._state_in_circuit_factory.num_ancilla_qubits
```

Comment:
I will note that the unit test coverage went down on PhaseEstimation with a number of the new additions, such as this, not having any unit test coverage. Can we do something to including testing of this newly added logic

Reply:
Actually I think these lines are not being tested because they are not needed. If the A and Q operator are given as circuits, the QAE algorithm uses the `PhaseEstimation` circuit from the library instead of the component. I'll remove this functionality (and type hints).